---
title: "Data process"
author: "Haipeng Sun"
date: "7/29/2022"
output: html_document
---

## library
```{r setup, message=FALSE, warning=FALSE}
library(readxl)
library(phyloseq)
library(qiime2R)
library(purrr)
library(furrr)
library(picante)
library(vegan)
library(MASS)
library(ggpubr)
library(rlist)
library(usedist)
library(tidyverse)
```

## settings
```{r}
# set path
path_rec <- "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/"
temp_out <- paste0(path_rec, "result/")
#metafile <- "MasterMetadata-CS-SW-20220623-hs.xlsx"
# set colors, labs and so on
#BirthMode_color <- c("VF"="#377EB8", "CSR"="#4DAF4A", "CS"="#E41A1C")
#Sex_labs <- c("F" = "Female", "M" = "Male")
#Sex_color <- c("F"="#e9a3c9", "M"="#a1d76a")
#responder_color <- c("Responder"="#d95f0e", "Nonresponder"="gray50")
#Age_color <- c("Week3"="#fdbe85", "Week6"="#fd8d3c", "Week9-14"="#e6550d", "Week18"="#a63603")
#BW_status_color <- c("Overweight"="#7570b3", "Normal"="#b3cde3","Underweight"="#66a61e")
```

## functions
```{r}

alpha_diversity_bootstrap <- function(asv_count_df, phylotree, depth, n=100){
  alpha_diversity_sample <- function(asv_count_df, phylotree, seed, depth){
      all_asv_count <- rep(asv_count_df$ASV, asv_count_df$count)
      set.seed(seed)
      boot_asv <- sample(all_asv_count, size = depth, replace = T)
      boot_dat <- table(boot_asv) %>% as.matrix() 
      colnames(boot_dat) <- "count"
      faith <- pd(t(boot_dat), phylotree, include.root = T)$PD
      observed <- nrow(boot_dat)
      shannon <- diversity(boot_dat, index = "shannon")
      evenness <- shannon/log(observed)
      data.frame(faith = faith, observed = observed, shannon = shannon, evenness = evenness)
  }
  if (sum(asv_count_df$count) < depth){
    c(faith = NA, observed = NA, shannon = NA, evenness = NA)
  }else{
    map_dfr(c(1:n), alpha_diversity_sample, 
                      asv_count_df = asv_count_df, phylotree = phylotree, 
                      depth = depth,
            .id = NULL) %>% colMeans()
    }
}

dist_to_long_tbl <- function(dist_matrix){
  dist_matrix %>% as.matrix() %>% as_tibble(rownames = "Sample1") %>%
      pivot_longer(-Sample1) %>% 
      rename(Sample2 = name, Distance = value) %>% 
      filter(Sample2 <= Sample1)
} 

cal_unifrac_dist_long <- function(rarified_psdata){
  ptree <- rarified_psdata@phy_tree
  ptree_new <- ape::multi2di(ptree, random = F)
  rarified_psdata@phy_tree <- ptree_new
  DM_uu <- UniFrac(rarified_psdata, weighted = FALSE)  
  DM_wu <- UniFrac(rarified_psdata, weighted = TRUE)
  dist_tbl_uu <- dist_to_long_tbl(DM_uu)
  dist_tbl_wu <- dist_to_long_tbl(DM_wu)
  dist_tbl <- inner_join(dist_tbl_uu, dist_tbl_wu, 
                         by = c("Sample1", "Sample2")) %>% 
    pivot_longer(cols = starts_with("Distance"),
                 names_to = "Metric") %>% 
    mutate(Metric = ifelse(Metric=="Distance.x", "uu","wu"))
  dist_tbl
}

cal_unifrac_dist_rarefy <- function(psdata, depth, rngseed){
    if(max(sample_sums(psdata)) < depth) return()
    psdata <- prune_samples(sample_sums(psdata) >= depth, psdata)
    rarified_psdata <- rarefy_even_depth(psdata, depth, replace = TRUE, verbose = FALSE, rngseed = rngseed)
    dist_tbl <- cal_unifrac_dist_long(rarified_psdata)
}

cal_unifrac_dist_rarefy_chunk <- function(psdata, depth, int_list){
  require(data.table)
  res_long <- future_map_dfr(int_list, cal_unifrac_dist_rarefy, 
                                psdata = psdata, depth = depth, 
                                .id = "N", .progress = F,
                                .options = furrr_options(seed=NULL))
  res_dt <- as.data.table(res_long %>% dplyr::select(Sample1, Sample2, Metric, value))
  res <- res_dt[, .(value = mean(value)), by = .(Sample1, Sample2, Metric)]
}

cal_unifrac_dist_rarefy_bootstrap <- function(psdata, depth, n, block_size = 100){
  if (n <= block_size){
    res <- cal_unifrac_dist_rarefy_chunk(psdata = psdata, depth = depth, c(1:n))
  }else{
    lst <- split(c(1:n), cut(1:n, breaks = ceiling(n/block_size)))
    res_lst <- lapply(lst, function(x){
      res_l <- cal_unifrac_dist_rarefy_chunk(psdata = psdata, depth = depth, x)
      res_l$value <- res_l$value * length(x)
      res_l
    })
    res_bind <- do.call(rbind, res_lst)
    res <- res_bind[, .(value = sum(value)/n), by = .(Sample1, Sample2, Metric)]
  }
  res
}
```


## 16S data process
### import into phyloseq
```{r}
phylo_all <- qza_to_phyloseq(features = paste0(path_rec,"05_merged/table_all.qza"),
                         taxonomy = paste0(path_rec,"06_taxa/taxonomy_dada_silva138.qza"),
                         tree =  paste0(path_rec,"05_merged/rooted_tree.qza"),
                         metadata = paste0(path_rec,"Mapping_all.txt"))
```

### metadata
```{r}
meta_all <- phylo_all@sam_data@.Data %>% as.data.frame()
names(meta_all) <- phylo_all@sam_data@names
meta_all$SampleID <- phylo_all@sam_data@row.names
meta_all$Seq_count <- sample_sums(phylo_all)

# add necessary information TBD

# add blanks info
meta_all$Sample_or_Blank <- ifelse(str_detect(meta_all$SampleID, "blank"),T,F)

# replace samdata in phyloseq
sample_data(phylo_all) <- sample_data(meta_all %>% column_to_rownames(var = "SampleID"))
```

### taxa
```{r}
taxa = tax_table(phylo_all@tax_table) %>% as.matrix() %>% as.data.frame() %>% rownames_to_column(var = "id")
taxa = taxa %>% mutate(ASV = str_c("asv_", 1:nrow(taxa)),
                       Kingdom = str_replace_na(Kingdom,""), Kingdom = str_replace(Kingdom,"d__","k_"),
                       Phylum = str_replace_na(Phylum,""), Phylum = str_c("p_",Phylum),
                       Class = str_replace_na(Class,""), Class = str_c("c_",Class),
                       Order = str_replace_na(Order,""), Order = str_c("o_",Order),
                       Family = str_replace_na(Family,""), Family = str_c("f_",Family),
                       Genus = str_replace_na(Genus,""), Genus = str_c("g_",Genus),
                       Species = str_replace_na(Species,""), Species = str_c("s_",Species),
                       taxa_long = str_c(Kingdom,Phylum,Class,Order,Family,Genus,Species,ASV)
                      )
```

### decontam
```{r }
library(decontam)
contamdf = isContaminant(phylo_all, method="prevalence",neg="Sample_or_Blank" )
contam_id = contamdf %>% filter(contaminant) %>% rownames()
phylo_decontam = prune_taxa(!(taxa_names(phylo_all) %in% contam_id), phylo_all)
taxa %>% filter(id %in% contam_id)
rm(contamdf, contam_id)
```

### filter unwanted taxa
```{r}
remove_id <- c(taxa %>% filter(str_detect(Kingdom, "Unassigned")) %>% pull(id),
               taxa %>% filter(str_detect(Kingdom, "Eukaryota")) %>% pull(id),
               taxa %>% filter(str_detect(taxa_long, "Chloroplast")) %>% pull(id),
               taxa %>% filter(str_detect(taxa_long, "Mitochondria")) %>% pull(id)
               )
phylo_decontam_clean <- prune_taxa(!(taxa_names(phylo_decontam) %in% remove_id), phylo_decontam)
rm(remove_id)
mean(sample_sums(phylo_all))
mean(sample_sums(phylo_decontam))
mean(sample_sums(phylo_decontam_clean))
sample_depth <- cbind(sample_sums(phylo_all), sample_sums(phylo_decontam),sample_sums(phylo_decontam_clean)) %>% 
  as.data.frame() %>% 
  rename(original=V1, decontamed=V2, decontam_cleaned=V3) %>%
  merge(., meta_all, by.x=0, by.y="SampleID") %>% 
  transmute(SampleID=Row.names, DNA_Conc_ng_uL, Country, Continent, 
            original, decontamed, decontam_cleaned) %>% 
  arrange(decontam_cleaned)
sample_depth %>% head(n = 50)
```

### rarefy depth
```{r }
ft <- otu_table(phylo_decontam_clean) %>% as.matrix() %>% as.data.frame() %>% t()
coverage_stat <- ft %>% as.data.frame() %>% 
  rownames_to_column(var = "SampleID") %>% 
  pivot_longer(cols = -SampleID, names_to = "id", values_to = "count") %>% 
  filter(count > 0) %>% group_by(SampleID) %>%
  summarise(n_seqs = sum(count),
            n_singles = sum(count==1),
            goods = 100*(1 - n_singles/n_seqs)) %>% 
  ungroup()

coverage_stat %>% ggplot(aes(x=n_seqs)) + 
  geom_histogram() + 
  labs(x="Depth", title = "Histograph of sequencing depth") 

coverage_stat %>% arrange(n_seqs) %>%
  ggplot(aes(x=1:nrow(.), y=n_seqs))+ 
  geom_point() + 
  labs(x = "", y="Depth", title = "Sequencing depth low to high")

coverage_stat %>%ggplot(aes(x=n_seqs, y=goods)) + 
  geom_point() + 
  geom_hline(yintercept = 99, color = "red") + 
  labs(x = "Depth", y = "Goods coverage", title = "Goods coverage")

rarecurve_data = rarecurve(ft, step = 50, tidy = T)

rarecurve_data %>% ggplot(aes(x=Sample, y=Species, group=Site)) + 
  geom_line(alpha = 0.2) + 
  scale_y_log10() +
  labs(x = "Depth", y = "Number of ASVs") + 
  coord_cartesian(xlim = c(0,10000)) 

```

### set depth to 2857
```{r }
depth = 2857
coverage_stat %>%ggplot(aes(x=n_seqs, y=goods)) + 
  geom_point() + 
  geom_hline(yintercept = 99, color = "red") + 
  geom_label(data = data.frame(n_seqs = c(60000,12000), 
                               goods = c(98, 80), 
                               label = c("99 Goods coverage", "Rarefy depth")), 
             aes(label = label)) + 
  geom_vline(xintercept = depth, color = "black") + 
  labs(x = "Depth", y = "Goods coverage", title = "Goods coverage")

rarecurve_data %>% ggplot(aes(x=Sample, y=Species, group=Site)) + 
  geom_line(alpha = 0.3) + 
  geom_vline(xintercept = depth, color = "black") + 
  geom_label(data = data.frame(Sample = c(2600), 
                               Species = c(900), 
                               Site = "",
                               label = c("Rarefy depth")), 
             aes(label = label)) + 
  scale_y_log10() +
  labs(x = "Depth", y = "Number of ASVs") + 
  coord_cartesian(xlim = c(0,5000)) 


phylo_decontam_clean_rf = rarefy_even_depth(phylo_decontam_clean, 
                                      sample.size = depth, 
                                      rngseed = 3, 
                                      replace = F)
rm(ft, coverage_stat)
```


## save data
```{r eval=FALSE, include=TRUE}
#save.image(file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/phylo_meta.Rdata")
load(file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/phylo_meta.Rdata")
```

## rarefy alpha diversity
```{r eval=FALSE, include=FALSE}
ft <- t((subset_samples(phylo_decontam_clean, !Sample_or_Blank))@otu_table@.Data)
ft_list <- ft %>% 
  as.data.frame() %>% rownames_to_column(var = "SampleID") %>%
  pivot_longer(-SampleID, names_to = "ASV", values_to = "count") %>%
  filter(count > 0) %>% split(., .$SampleID)
phylotree <- phylo_decontam_clean@phy_tree

plan(multisession)
system.time(aa <- future_map_dfr(ft_list, alpha_diversity_bootstrap, 
                                phylotree = phylotree, depth = depth, n = 100, 
                                .id = "SampleID", .progress = F,
                                .options = furrr_options(seed=NULL)))
#user  system elapsed 
#38.992   1.763 547.710


alpha_boot <- aa %>% filter(!is.na(faith)) %>%
  dplyr::select(SampleID, Faith_PD=faith, Observed_ASVs=observed, 
                Shannon_Index=shannon, Pielou_evenness=evenness)

a1 <- pd(t(phylo_decontam_clean_rf@otu_table@.Data), phylo_decontam_clean_rf@phy_tree, include.root = F)
a2 <- estimate_richness(phylo_decontam_clean_rf, measures="Shannon")
alpha_rftb <- merge(a1, a2, by=0)
names(alpha_rftb) <- c("SampleID","Faith_PD", "Observed_ASVs","Shannon_Index")
alpha_rftb$Pielou_evenness <- alpha_rftb$Shannon_Index/log(alpha_rftb$Observed_ASVs)


#save(alpha_boot, alpha_rftb,file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/alpha.Rdata")
```

### compare
```{r}
load(file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/alpha.Rdata")
alpha_test <- inner_join(alpha_rftb, alpha_boot, by = "SampleID")
alpha_test %>% pivot_longer(-SampleID, names_to = c("Metrics", "Method"), names_sep = "\\.") %>% 
  mutate(Method = ifelse(Method == "x", "Rarefy_table", "Bootstrap")) %>% 
  pivot_wider(c(SampleID, Metrics), names_from = Method, values_from = value) %>% 
  ggplot(aes(x = Rarefy_table, y = Bootstrap)) + 
    geom_point() + 
    geom_smooth(method = "lm") + 
    stat_regline_equation(label.y.npc = 0.8) + 
    stat_cor(label.y.npc = 0.9) + 
    facet_wrap(~ Metrics, nrow = 2, scales = "free")
```

## rarefy beta diversity
```{r}

psdata <- subset_samples(phylo_decontam_clean, !Sample_or_Blank)
plan(multisession)
system.time(unif_two <- cal_unifrac_dist_rarefy_bootstrap(psdata = psdata, depth = depth, n = 100, block_size = 100))
#n = 10 data.table = 10
#    user   system  elapsed 
#  39.154    3.282 1003.004 

#n = 100 data.table = 10
#    user   system  elapsed 
# 646.566   37.762 7630.126 

DM_uu <- pivot_to_numeric_matrix(unif_two %>% filter(Metric == "uu"), Sample1, Sample2, value) %>% as.dist()
DM_wu <- pivot_to_numeric_matrix(unif_two %>% filter(Metric == "wu"), Sample1, Sample2, value) %>% as.dist()

ft = t(psdata@otu_table@.Data)
system.time(DM_bray <- avgdist(ft,dmethod = "bray", sample = depth, iterations = 100))
# user   system  elapsed 
# 2782.295   27.415 2808.332
system.time(DM_jaccard <- avgdist(ft,dmethod = "jaccard", sample = depth, iterations = 100))
# user   system  elapsed 
# 2782.295   27.415 2808.332 


#save(DM_uu,file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/beta_DM_uu.RData")
#save(DM_wu,file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/beta_DM_wu.RData")
#save(DM_bray,file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/beta_DM_bray.RData")
#save(DM_jaccard,file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/beta_DM_jaccard.RData")

```













