---
title: "Data process"
author: "Haipeng Sun"
date: "7/29/2022"
output: html_document
---

## library
```{r setup, message=FALSE, warning=FALSE}
library(readxl)
library(phyloseq)
library(qiime2R)
library(purrr)
library(furrr)
library(picante)
library(vegan)
library(MASS)
library(ggpubr)
library(rlist)
library(usedist)
library(ggtext)
library(glue)
library(RColorBrewer)
library(tidyverse)
```

## settings
```{r}
# set path
path_rec <- "~/Library/CloudStorage/Dropbox/Rutgers/20220606_RUN41_milk/Data/"
temp_out <- paste0(path_rec, "result/")
# set colors, labs and so on
col_vector = c(brewer.pal(12, "Set3"), brewer.pal(12, "Paired"), 
               brewer.pal(8, "Set2"), brewer.pal(7, "Accent")) %>% unique()
names(col_vector) = col_vector
#g <- ggplot(data.frame(name=col_vector, y = rep(1, length(col_vector)))) + geom_col(aes(x=name, y=y, color=name, fill=name)) + scale_color_manual(values = col_vector, aesthetics = c("color","fill")) + theme(axis.text.x = element_text(angle=90, vjust = 0.5, hjust = 1))
#ggsave("~/Desktop/col_vector.pdf", plot = g, device = "pdf")
#rm(g)

Continent_color <- c("AFRICA"="#7fc97f", "AMERICA"="#beaed4", "EUROPE"="#fdc086")
Country_color <- c("Equatorial Guinea"="#B2DF8A","Kenya"="#33A02C","Senegal"="#A6D854","South Sudan"="#7FC97F","Tanzania"="#66C2A5",
                   "Ecuador"="#CAB2D6","ElSalvador"="#6A3D9A","Mexico"="#BC80BD","Peru"="#8DA0CB","Puerto Rico"="#A6CEE3","United States"="#FB9A99",
                   "Austria"="#E5C494","Germany"="#BF5B17","Netherlands"="#FF7F00","Norway"="#FB8072", "Spain"="#FDB462") 

Country_font_color = c("Equatorial Guinea"="#1b9e77", "Kenya"="#1b9e77", "Senegal"="#1b9e77", "South Sudan"="#1b9e77", "Tanzania"="#1b9e77",
                       "Ecuador"="#7570b3", "ElSalvador"="#7570b3", "Mexico"="#7570b3", "Peru"="#7570b3", "Puerto Rico"="#7570b3", "United States"="#7570b3",
                       "Austria"="#d95f02", "Germany"="#d95f02", "Netherlands"="#d95f02", "Norway"="#d95f02", "Spain"="#d95f02")
Country_labels = glue("<span style='color:{Country_font_color}'>{names(Country_font_color)}</span>")
names(Country_labels) <- names(Country_font_color)

#ggplot(data.frame(country = factor(names(Country_color), levels = names(Country_color), ordered = T), y = rep(1, 16))) + 
#  geom_col(aes(x=country, y=y, color=country, fill=country)) + scale_color_manual(values = Country_color, aesthetics = c("color","fill")) + 
#  scale_x_discrete(labels=Country_labels) + theme(axis.text.x = element_markdown(size=12,face='bold',angle=90,vjust = 0.5, hjust = 1))
```

## functions
```{r}
alpha_diversity_bootstrap <- function(asv_count_df, phylotree, depth, n=100){
  alpha_diversity_sample <- function(asv_count_df, phylotree, seed, depth){
      all_asv_count <- rep(asv_count_df$ASV, asv_count_df$count)
      set.seed(seed)
      boot_asv <- sample(all_asv_count, size = depth, replace = T)
      boot_dat <- table(boot_asv) %>% as.matrix() 
      colnames(boot_dat) <- "count"
      faith <- pd(t(boot_dat), phylotree, include.root = T)$PD
      observed <- nrow(boot_dat)
      shannon <- diversity(boot_dat, index = "shannon")
      evenness <- shannon/log(observed)
      data.frame(faith = faith, observed = observed, shannon = shannon, evenness = evenness)
  }
  if (sum(asv_count_df$count) < depth){
    c(faith = NA, observed = NA, shannon = NA, evenness = NA)
  }else{
    map_dfr(c(1:n), alpha_diversity_sample, 
                      asv_count_df = asv_count_df, phylotree = phylotree, 
                      depth = depth,
            .id = NULL) %>% colMeans()
    }
}

dist_to_long_tbl <- function(dist_matrix){
  dist_matrix %>% as.matrix() %>% as_tibble(rownames = "Sample1") %>%
      pivot_longer(-Sample1) %>% 
      rename(Sample2 = name, Distance = value) %>% 
      filter(Sample2 <= Sample1)
} 

cal_unifrac_dist_long <- function(rarified_psdata){
  ptree <- rarified_psdata@phy_tree
  ptree_new <- ape::multi2di(ptree, random = F)
  rarified_psdata@phy_tree <- ptree_new
  DM_uu <- UniFrac(rarified_psdata, weighted = FALSE)  
  DM_wu <- UniFrac(rarified_psdata, weighted = TRUE)
  dist_tbl_uu <- dist_to_long_tbl(DM_uu)
  dist_tbl_wu <- dist_to_long_tbl(DM_wu)
  dist_tbl <- inner_join(dist_tbl_uu, dist_tbl_wu, 
                         by = c("Sample1", "Sample2")) %>% 
    pivot_longer(cols = starts_with("Distance"),
                 names_to = "Metric") %>% 
    mutate(Metric = ifelse(Metric=="Distance.x", "uu","wu"))
  dist_tbl
}

cal_unifrac_dist_rarefy <- function(psdata, depth, rngseed){
    if(max(sample_sums(psdata)) < depth) return()
    psdata <- prune_samples(sample_sums(psdata) >= depth, psdata)
    rarified_psdata <- rarefy_even_depth(psdata, depth, replace = TRUE, verbose = FALSE, rngseed = rngseed)
    dist_tbl <- cal_unifrac_dist_long(rarified_psdata)
}

cal_unifrac_dist_rarefy_chunk <- function(psdata, depth, int_list){
  require(data.table)
  res_long <- future_map_dfr(int_list, cal_unifrac_dist_rarefy, 
                                psdata = psdata, depth = depth, 
                                .id = "N", .progress = F,
                                .options = furrr_options(seed=NULL))
  res_dt <- as.data.table(res_long %>% dplyr::select(Sample1, Sample2, Metric, value))
  res <- res_dt[, .(value = mean(value)), by = .(Sample1, Sample2, Metric)]
}

cal_unifrac_dist_rarefy_bootstrap <- function(psdata, depth, n, block_size = 100){
  if (n <= block_size){
    res <- cal_unifrac_dist_rarefy_chunk(psdata = psdata, depth = depth, c(1:n))
  }else{
    lst <- split(c(1:n), cut(1:n, breaks = ceiling(n/block_size)))
    res_lst <- lapply(lst, function(x){
      res_l <- cal_unifrac_dist_rarefy_chunk(psdata = psdata, depth = depth, x)
      res_l$value <- res_l$value * length(x)
      res_l
    })
    res_bind <- do.call(rbind, res_lst)
    res <- res_bind[, .(value = sum(value)/n), by = .(Sample1, Sample2, Metric)]
  }
  res
}
```

## 16S data process
### import into phyloseq
```{r}
phylo_all <- qza_to_phyloseq(features = paste0(path_rec,"05_merged/table_all.qza"),
                         taxonomy = paste0(path_rec,"06_taxa/taxonomy_dada_silva138.qza"),
                         tree =  paste0(path_rec,"05_merged/rooted_tree.qza"),
                         metadata = paste0(path_rec,"Mapping_all.txt"))
```

### metadata
```{r}
meta_all <- phylo_all@sam_data@.Data %>% as.data.frame()
names(meta_all) <- phylo_all@sam_data@names
meta_all$SampleID <- phylo_all@sam_data@row.names
meta_all$Seq_count <- sample_sums(phylo_all)

# add necessary information TBD
#1 heat inactivation compare info
heated_sample <- meta_all %>% filter(heat_inactivated == "Y", !str_detect(SampleID, "SARS")) %>% pull(Original_SampleID)
meta_all <- meta_all %>% 
  mutate(heat_compare = ifelse(Original_SampleID %in% heated_sample, "Y", "N"))
rm(heated_sample)
#2 add blanks info
meta_all$Sample_or_Blank <- ifelse(str_detect(meta_all$SampleID, "blank"),T,F)
#3 add different country group
meta_all <- meta_all %>% mutate(Conti_grp = case_when(Continent == "AFRICA" ~ "Africa",
                                              Continent == "EUROPE" ~ "Europe",
                                              Country == "United States" ~ "United States",
                                              Country == "Puerto Rico" ~ "Puerto Rico",
                                              Continent == "AMERICA" & !Country %in% c("United States","Puerto Rico") ~ "Central and South America"))
#4 mark if send for ITS
temp <- read_excel("~/Dropbox/Rutgers/20220304_Milk_inventory/ITS_upside_removed/Metadata_global_milk_platemap.xlsx", sheet = "Global_milk_sample_mapping")
meta_all <- meta_all %>% mutate(ITS = ifelse(SampleID %in% temp$TubeID, "Y", "N"))
rm(temp)
#4 corrections
meta_all <- meta_all %>% mutate(Country = str_replace(Country, "Guinea Ecuatorial", "Equatorial Guinea"))

# replace samdata in phyloseq
sample_data(phylo_all) <- sample_data(meta_all %>% column_to_rownames(var = "SampleID"))
```

### taxa
```{r}
taxa = tax_table(phylo_all@tax_table) %>% as.matrix() %>% as.data.frame() %>% rownames_to_column(var = "id")
taxa = taxa %>% mutate(ASV = str_c("asv_", 1:nrow(taxa)),
                       Kingdom = str_replace_na(Kingdom,""), Kingdom = str_replace(Kingdom,"d__","k_"),
                       Phylum = str_replace_na(Phylum,""), Phylum = str_c("p_",Phylum),
                       Class = str_replace_na(Class,""), Class = str_c("c_",Class),
                       Order = str_replace_na(Order,""), Order = str_c("o_",Order),
                       Family = str_replace_na(Family,""), Family = str_c("f_",Family),
                       Genus = str_replace_na(Genus,""), Genus = str_c("g_",Genus),
                       Species = str_replace_na(Species,""), Species = str_c("s_",Species),
                       taxa_long = str_c(Kingdom,Phylum,Class,Order,Family,Genus,Species,ASV,sep = ";")
                      )
```

### decontam
```{r }
library(decontam)
a0 <- isContaminant(phylo_all, method="prevalence",neg="Sample_or_Blank" )
a1 <- isContaminant(phylo_all, method="combined", conc="PCRConc", neg="Sample_or_Blank")
a2 <- isContaminant(phylo_all, method="minimum", conc="PCRConc", neg="Sample_or_Blank")
a3 <- isContaminant(phylo_all, method="either", conc="PCRConc", neg="Sample_or_Blank")
a4 <- isContaminant(phylo_all, method="both", conc="PCRConc", neg="Sample_or_Blank")

prev <- a0 %>% filter(contaminant) %>% rownames()
combined <- a1 %>% filter(contaminant) %>% rownames()
minimum <- a2 %>% filter(contaminant) %>% rownames()
either <- a3 %>% filter(contaminant) %>% rownames()
both <- a4 %>% filter(contaminant) %>% rownames()

lst <- list(prev, combined, minimum, either, both)
names(lst) <- c("prev", "combined", "minimum", "either", "both")
gplots::venn(lst)

temp <- taxa %>% filter(id %in% unique(unlist(lst))) %>% 
  mutate(prev = ifelse(id %in% prev, "Y", "N"),
         combined = ifelse(id %in% combined, "Y", "N"),
         minimum = ifelse(id %in% minimum, "Y", "N"),
         either = ifelse(id %in% either, "Y", "N"),
         both = ifelse(id %in% both, "Y", "N")) %>% 
  select(taxa_long, prev, combined, minimum, either, both)
write.csv(temp, file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/decontam_details.csv", quote = F, row.names = F)

# use pre method to decontam
contamdf <- a0
contam_id <- contamdf %>% filter(contaminant) %>% rownames()
phylo_decontam <- prune_taxa(!(taxa_names(phylo_all) %in% contam_id), phylo_all)
rm(contamdf, contam_id, a0,  a1, a2, a3, a4, prev, combined, minimum, either, both, lst, temp)
```

### filter unwanted taxa
```{r}
remove_id <- c(taxa %>% filter(str_detect(Kingdom, "Unassigned")) %>% pull(id),
               taxa %>% filter(str_detect(Kingdom, "Eukaryota")) %>% pull(id),
               taxa %>% filter(str_detect(taxa_long, "Chloroplast")) %>% pull(id),
               taxa %>% filter(str_detect(taxa_long, "Mitochondria")) %>% pull(id)
               )
phylo_decontam_clean <- prune_taxa(!(taxa_names(phylo_decontam) %in% remove_id), phylo_decontam)
rm(remove_id)
mean(sample_sums(phylo_all))
mean(sample_sums(phylo_decontam))
mean(sample_sums(phylo_decontam_clean))
sample_depth <- cbind(sample_sums(phylo_all), sample_sums(phylo_decontam),sample_sums(phylo_decontam_clean)) %>% 
  as.data.frame() %>% 
  rename(original=V1, decontamed=V2, decontam_cleaned=V3) %>%
  merge(., meta_all, by.x=0, by.y="SampleID") %>% 
  transmute(SampleID=Row.names, DNA_Conc_ng_uL, Country, Continent,
            original, decontamed, decontam_cleaned) %>% 
  arrange(decontam_cleaned)
sample_depth %>% head(n = 50)

write.csv(sample_depth, file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/sample_clean_depth.csv", quote = F, row.names = F)

my_summarize = function(x){
  data.frame(
             #N = n(),
             Min = min(x,na.rm = T),
             Q5 = quantile(x, probs = 0.05, na.rm = T),
             Q10 = quantile(x, probs = 0.1, na.rm = T),
             Q25 = quantile(x, probs = 0.25, na.rm = T),
             Med = median(x, na.rm = T),
             Mean = mean(x, na.rm = T),
             Q75 = quantile(x, probs = 0.75, na.rm = T),
             Q95 = quantile(x, probs = 0.95, na.rm = T),
             Max = max(x,na.rm = T)
             )
}

a <- lapply(c("original", "decontamed", "decontam_cleaned"), function(x){
  temp <- my_summarize(sample_depth %>% pull(x))
  temp
}) %>% do.call(rbind, .)
rownames(a) <- c("original", "decontamed", "decontam_cleaned")

write.csv(a, file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/sample_clean_depth_summary.csv", quote = F, row.names = T)


```

### rarefy depth
```{r }
ft <- otu_table(phylo_decontam_clean) %>% as.matrix() %>% as.data.frame() %>% t()
coverage_stat <- ft %>% as.data.frame() %>% 
  rownames_to_column(var = "SampleID") %>% 
  pivot_longer(cols = -SampleID, names_to = "id", values_to = "count") %>% 
  filter(count > 0) %>% group_by(SampleID) %>%
  summarise(n_seqs = sum(count),
            n_singles = sum(count==1),
            goods = 100*(1 - n_singles/n_seqs)) %>% 
  ungroup()

#rarecurve_data = rarecurve(ft, step = 50, tidy = T)
#save(rarecurve_data, file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/rarecurve.Rdata")
load(file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/rarecurve.Rdata")

coverage_stat %>% ggplot(aes(x=n_seqs)) + 
  geom_histogram() + 
  labs(x="Depth", title = "Histograph of sequencing depth") 

coverage_stat %>% arrange(n_seqs) %>%
  ggplot(aes(x=1:nrow(.), y=n_seqs))+ 
  geom_point() + 
  labs(x = "", y="Depth", title = "Sequencing depth low to high")

coverage_stat %>%ggplot(aes(x=n_seqs, y=goods)) + 
  geom_point() + 
  geom_hline(yintercept = 99, color = "red") + 
  labs(x = "Depth", y = "Goods coverage", title = "Goods coverage")

rarecurve_data %>% ggplot(aes(x=Sample, y=Species, group=Site)) + 
  geom_line(alpha = 0.2) + 
  scale_y_log10() +
  labs(x = "Depth", y = "Number of ASVs") + 
  coord_cartesian(xlim = c(0,10000)) 

### set depth to 2857
depth = 2857
g <- coverage_stat %>%ggplot(aes(x=n_seqs, y=goods)) + 
  geom_point() + 
  geom_hline(yintercept = 99, color = "red") + 
  geom_label(data = data.frame(n_seqs = c(60000,12000), 
                               goods = c(98, 80), 
                               label = c("99 Goods coverage", "Rarefy depth")), 
             aes(label = label)) + 
  geom_vline(xintercept = depth, color = "black") + 
  labs(x = "Depth", y = "Goods coverage", title = "Goods coverage")
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/Goods_coverage_depth.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

g <- rarecurve_data %>% ggplot(aes(x=Sample, y=Species, group=Site)) + 
  geom_line(alpha = 0.3) + 
  geom_vline(xintercept = depth, color = "black") + 
  geom_label(data = data.frame(Sample = c(2600), 
                               Species = c(900), 
                               Site = "",
                               label = c("Rarefy depth")), 
             aes(label = label)) + 
  scale_y_log10() +
  labs(x = "Depth", y = "Number of ASVs") + 
  coord_cartesian(xlim = c(0,5000)) 
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/depth_rarefy.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

phylo_decontam_clean_rf = rarefy_even_depth(phylo_decontam_clean, 
                                      sample.size = depth, 
                                      rngseed = 3, 
                                      replace = F)
rm(ft, coverage_stat, g)
```

## save data
```{r eval=FALSE, include=TRUE}
#save.image(file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/phylo_meta.Rdata")
load(file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/phylo_meta.Rdata")
```

## rarefy alpha diversity
```{r eval=FALSE, include=TRUE}
ft <- t((subset_samples(phylo_decontam_clean, !Sample_or_Blank))@otu_table@.Data)
ft_list <- ft %>% 
  as.data.frame() %>% rownames_to_column(var = "SampleID") %>%
  pivot_longer(-SampleID, names_to = "ASV", values_to = "count") %>%
  filter(count > 0) %>% split(., .$SampleID)
phylotree <- phylo_decontam_clean@phy_tree

plan(multisession)
system.time(aa <- future_map_dfr(ft_list, alpha_diversity_bootstrap, 
                                phylotree = phylotree, depth = depth, n = 100, 
                                .id = "SampleID", .progress = F,
                                .options = furrr_options(seed=NULL)))
#user  system elapsed 
#38.992   1.763 547.710
alpha_boot <- aa %>% filter(!is.na(faith)) %>%
  dplyr::select(SampleID, Faith_PD=faith, Observed_ASVs=observed, 
                Shannon_Index=shannon, Pielou_evenness=evenness)

a1 <- pd(t(phylo_decontam_clean_rf@otu_table@.Data), phylo_decontam_clean_rf@phy_tree, include.root = F)
a2 <- estimate_richness(phylo_decontam_clean_rf, measures="Shannon")
alpha_rftb <- merge(a1, a2, by=0)
names(alpha_rftb) <- c("SampleID","Faith_PD", "Observed_ASVs","Shannon_Index")
alpha_rftb$Pielou_evenness <- alpha_rftb$Shannon_Index/log(alpha_rftb$Observed_ASVs)

#save(alpha_boot, alpha_rftb,file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/alpha.Rdata")
```

### compare
```{r}
load(file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/alpha.Rdata")
alpha_test <- inner_join(alpha_rftb, alpha_boot, by = "SampleID")
alpha_test %>% pivot_longer(-SampleID, names_to = c("Metrics", "Method"), names_sep = "\\.") %>% 
  mutate(Method = ifelse(Method == "x", "Rarefy_table", "Bootstrap")) %>% 
  pivot_wider(c(SampleID, Metrics), names_from = Method, values_from = value) %>% 
  ggplot(aes(x = Rarefy_table, y = Bootstrap)) + 
    geom_point() + 
    geom_smooth(method = "lm") + 
    stat_regline_equation(label.y.npc = 0.8) + 
    stat_cor(label.y.npc = 0.9) + 
    facet_wrap(~ Metrics, nrow = 2, scales = "free")
```

## rarefy beta diversity
```{r eval=FALSE, include=TRUE}
psdata <- subset_samples(phylo_decontam_clean, !Sample_or_Blank)
plan(multisession)
system.time(unif_two <- cal_unifrac_dist_rarefy_bootstrap(psdata = psdata, depth = depth, n = 100, block_size = 100))
#n = 10 data.table = 10
#    user   system  elapsed 
#  39.154    3.282 1003.004 
#n = 100 data.table = 10
#    user   system  elapsed 
# 646.566   37.762 7630.126 
DM_uu <- pivot_to_numeric_matrix(unif_two %>% filter(Metric == "uu"), Sample1, Sample2, value) %>% as.dist()
DM_wu <- pivot_to_numeric_matrix(unif_two %>% filter(Metric == "wu"), Sample1, Sample2, value) %>% as.dist()

ft = t(psdata@otu_table@.Data)
system.time(DM_bray <- avgdist(ft,dmethod = "bray", sample = depth, iterations = 100))
# user   system  elapsed 
# 2782.295   27.415 2808.332
system.time(DM_jaccard <- avgdist(ft,dmethod = "jaccard", sample = depth, iterations = 100))
# user   system  elapsed 
# 2782.295   27.415 2808.332 

#save(DM_uu,file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/beta_DM_uu.RData")
#save(DM_wu,file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/beta_DM_wu.RData")
#save(DM_bray,file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/beta_DM_bray.RData")
#save(DM_jaccard,file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/beta_DM_jaccard.RData")
```

## data summary
### by country/project
```{r}
dat <- phylo_decontam_clean@sam_data@.Data %>% as.data.frame()
names(dat) <- phylo_decontam_clean@sam_data@names
dat$SampleID <- phylo_decontam_clean@sam_data@row.names
dat$Seq_count_clean <- sample_sums(phylo_decontam_clean) 
dat <- dat %>% filter(!is.na(Country), Seq_count_clean >= depth)

a <- dat %>% select(SampleID, Extraction_method, heat_inactivated, Country, Project) %>% 
  group_by(Extraction_method, heat_inactivated, Country, Project) %>% 
  summarise(N = n()) %>% ungroup() %>% 
  pivot_wider(id_cols = c(Project, Extraction_method, heat_inactivated), 
              names_from = "Country",
              values_from = "N") 
write.csv(a, file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/by_project_N.csv",
          row.names = FALSE)

a <- dat %>% filter(!is.na(TimePoint)) %>% 
  select(SampleID, Project, TimePoint) %>% 
  group_by(Project, TimePoint) %>% 
  summarise(N = n()) %>% ungroup() %>% 
  pivot_wider(id_cols = Project, names_from = "TimePoint", values_from = "N")
write.csv(a, file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/circadian_samples_by_project_N.csv",
          row.names = FALSE)
# decided to include 1200 samples for global project
#meta_global<- rbind(dat %>% filter(!is.na(Country), Project %in% c("Global", "Upside", "Milkbank")),
#           dat %>% filter(!is.na(Country), Project=="PR") %>% filter(TimePoint == 1200))
#a <- meta_global %>% group_by(Country) %>% summarise(N = n()) %>% ungroup() 
#write.csv(a, file = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/Global_samples_by_Country.csv",
#          row.names = FALSE)
```

### correlations 
```{r}
cor(dat %>% select( DNA_Conc_ng_uL, PCRConc, Seq_count, Seq_count_clean), use = "complete.obs")
```

### effect of heat inactivate
```{r}
dat <- phylo_decontam_clean@sam_data@.Data %>% as.data.frame()
names(dat) <- phylo_decontam_clean@sam_data@names
dat$SampleID <- phylo_decontam_clean@sam_data@row.names
dat$Seq_count_clean <- sample_sums(phylo_decontam_clean)
dat <- dat %>% filter(!is.na(Country), heat_compare == "Y")

g <- ggplot(dat, aes(heat_inactivated, Seq_count_clean)) + 
  geom_boxplot(aes(color = heat_inactivated)) + 
  geom_point(aes(color = heat_inactivated)) + 
  geom_path(aes(group = Original_SampleID), alpha = 0.6, color = "gray") +
  stat_compare_means(paired = TRUE) + 
  scale_x_discrete(labels = c("N"="No heat", "Y"="Heat inactivated")) + 
  labs(x="", y="Clean sequence count") + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold',angle=0),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/heat_inactivation_on_clean_reads_count.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

g <- ggplot(dat, aes(heat_inactivated, DNA_Conc_ng_uL)) + 
  geom_boxplot(aes(color = heat_inactivated)) + 
  geom_point(aes(color = heat_inactivated)) + 
  geom_path(aes(group = Original_SampleID), alpha = 0.6, color = "gray") +
  stat_compare_means(paired = TRUE) + 
  scale_x_discrete(labels = c("N"="No heat", "Y"="Heat inactivated")) + 
  labs(x="", y=expression(bold("DNA concentration (ng/"*mu*"L)"))) + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold',angle=0),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/heat_inactivation_on_DNAconc.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

g <- ggplot(dat, aes(heat_inactivated, PCRConc)) + 
  geom_boxplot(aes(color = heat_inactivated)) + 
  geom_point(aes(color = heat_inactivated)) + 
  geom_path(aes(group = Original_SampleID), alpha = 0.6, color = "gray") +
  stat_compare_means(paired = TRUE) + 
  scale_x_discrete(labels = c("N"="No heat", "Y"="Heat inactivated")) + 
  labs(x="", y=expression(bold("PCR concentration (ng/"*mu*"L)"))) + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold',angle=0),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/heat_inactivation_on_PCRconc.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

# individual
wdat <- dat %>% 
              select(Original_SampleID, heat_inactivated, DNA_Conc_ng_uL, PCRConc, Seq_count_clean) %>% 
              pivot_wider(id_cols = Original_SampleID, 
                          names_from = heat_inactivated, names_glue = "{.value}_{heat_inactivated}",
                          values_from = c(DNA_Conc_ng_uL, PCRConc, Seq_count_clean)) 

g <- ggplot(wdat, aes(Seq_count_clean_N, Seq_count_clean_Y)) + 
  geom_point() + 
  geom_smooth(method = "lm") + 
  stat_regline_equation(label.x.npc = 0.6,label.y.npc = 0.05) + 
  stat_cor(label.x.npc = 0.6, label.y.npc = 0.01) + 
  scale_x_log10() + 
  scale_y_log10() +  
  labs(x="No heat", y="Heat inactivated", title = "Clean sequence count") + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold',angle=0),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/heat_inactivation_on_clean_reads_count_individual.pdf",
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

g <- ggplot(wdat, aes(DNA_Conc_ng_uL_N, DNA_Conc_ng_uL_Y)) + 
  geom_point() + 
  geom_smooth(method = "lm") + 
  stat_regline_equation(label.x.npc = 0.6,label.y.npc = 0.05) + 
  stat_cor(label.x.npc = 0.6, label.y.npc = 0.01) + 
  scale_x_log10() + 
  scale_y_log10() +  
  labs(x="No heat", y="Heat inactivated", title = "DNA concentration") + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold',angle=0),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/heat_inactivation_on_DNAconc_individual.pdf",
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

g <- ggplot(wdat, aes(PCRConc_N, PCRConc_Y)) + 
  geom_point() + 
  geom_smooth(method = "lm") + 
  stat_regline_equation(label.x.npc = 0.6,label.y.npc = 0.05) + 
  stat_cor(label.x.npc = 0.6, label.y.npc = 0.01) + 
  scale_x_log10() + 
  scale_y_log10() +  
  labs(x="No heat", y="Heat inactivated", title = "PCR concentration") + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold',angle=0),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/heat_inactivation_on_PCRconc_individual.pdf",
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)
```

### effect of extraction method
```{r}
# to reduce confounder only use US samples
dat <- phylo_decontam_clean@sam_data@.Data %>% as.data.frame()
names(dat) <- phylo_decontam_clean@sam_data@names
dat$SampleID <- phylo_decontam_clean@sam_data@row.names
dat$Seq_count_clean <- sample_sums(phylo_decontam_clean) 
dat <- dat %>% filter(Country == "United States", Seq_count_clean >= depth)

g <- ggplot(dat, aes(Extraction_method, Seq_count_clean)) + 
  geom_boxplot(aes(color = Extraction_method)) + 
  geom_point(aes(color = Extraction_method), position = position_dodge2(width = 0.2)) + 
  stat_compare_means() + 
  labs(x="", y="Clean sequence count") + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold',angle=0),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/extract_method_on_clean_reads_count_US.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

g <- ggplot(dat, aes(Extraction_method, DNA_Conc_ng_uL)) + 
  geom_boxplot(aes(color = Extraction_method)) + 
  geom_point(aes(color = Extraction_method), position = position_dodge2(width = 0.2)) + 
  stat_compare_means() + 
  scale_y_log10() + 
  labs(x="", y=expression(bold("DNA concentration (ng/"*mu*"L)"))) + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold',angle=0),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/extract_method_on_DNAconc_US.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

g <- ggplot(dat, aes(Extraction_method, PCRConc)) + 
  geom_boxplot(aes(color = Extraction_method)) + 
  geom_point(aes(color = Extraction_method), position = position_dodge2(width = 0.2)) + 
  stat_compare_means() + 
  labs(x="", y=expression(bold("PCR concentration (ng/"*mu*"L)"))) + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold',angle=0),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/extract_method_on_PCRconc_US.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)
```

### depth by country
```{r}
dat <- phylo_decontam_clean@sam_data@.Data %>% as.data.frame()
names(dat) <- phylo_decontam_clean@sam_data@names
dat$SampleID <- phylo_decontam_clean@sam_data@row.names
dat$Seq_count_clean <- sample_sums(phylo_decontam_clean)
dat <- dat %>% filter(!is.na(Country), Seq_count_clean >= depth, heat_inactivated == "N") %>% 
  mutate(Country = factor(Country,
                          levels = c("Guinea Ecuatorial", "Kenya", "Senegal", "South Sudan", "Tanzania",
                                     "Ecuador", "ElSalvador", "Mexico", "Peru", "Puerto Rico", "United States",
                                     "Austria", "Germany", "Netherlands", "Norway", "Spain"),
                          ordered = T))

g <- ggplot(dat, aes(Country, Seq_count_clean)) + 
  geom_boxplot(aes(color = Country, fill = Country), 
               outlier.alpha = NULL, alpha = 0.8) + 
  geom_point(aes(color = Country)) +
  scale_color_manual(values = Country_color, aesthetics = c("color","fill")) + 
  labs(x = "", y = "Clean sequence count") + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold',angle=90, vjust = 0.5, hjust = 1),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/clean_reads_count_by_country.pdf", device = cairo_pdf, 
         plot = g, width = 7, height = 5, units = "in", dpi = 300)
```

### depth vs DNA concentration
```{r}
dat <- phylo_decontam_clean@sam_data@.Data %>% as.data.frame()
names(dat) <- phylo_decontam_clean@sam_data@names
dat$SampleID <- phylo_decontam_clean@sam_data@row.names
dat$Seq_count_clean <- sample_sums(phylo_decontam_clean)
dat <- dat %>% filter(!is.na(Country), Seq_count_clean >= depth)

g <- ggplot(dat, aes(DNA_Conc_ng_uL, Seq_count_clean, color = heat_inactivated)) + 
  geom_point() + 
  scale_x_log10() + 
  scale_y_log10() +
  geom_smooth(method = "lm") + 
  stat_regline_equation(label.x.npc = 0.6,label.y.npc = 0.15) + 
  stat_cor(label.x.npc = 0.6, label.y.npc = 0.05) + 
  labs(x = expression(bold("DNA concentration (ng/"*mu*"L)")), y = "Clean sequence count") + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     #legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold'),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/clean_reads_count_vs_DNAconc_by_heat.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

g <- ggplot(dat, aes(DNA_Conc_ng_uL, Seq_count_clean, color = Extraction_method)) + 
  geom_point() + 
  scale_x_log10() + 
  scale_y_log10() +
  geom_smooth(method = "lm") + 
  stat_regline_equation(label.x.npc = 0.6,label.y.npc = 0.15) + 
  stat_cor(label.x.npc = 0.6, label.y.npc = 0.05) + 
  labs(x = expression(bold("DNA concentration (ng/"*mu*"L)")), y = "Clean sequence count") + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     #legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold'),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/clean_reads_count_vs_DNAconc_by_extract_method.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)
```

### depth vs PCR concentration
```{r}
dat <- phylo_decontam_clean@sam_data@.Data %>% as.data.frame()
names(dat) <- phylo_decontam_clean@sam_data@names
dat$SampleID <- phylo_decontam_clean@sam_data@row.names
dat$Seq_count_clean <- sample_sums(phylo_decontam_clean)
dat <- dat %>% filter(!is.na(Country), Seq_count_clean >= depth)

g <- ggplot(dat, aes(PCRConc, Seq_count_clean, color = heat_inactivated)) + 
  geom_point() + 
  scale_x_log10() + 
  scale_y_log10() +
  geom_smooth(method = "lm") + 
  stat_regline_equation(label.x.npc = 0.6,label.y.npc = 0.15) + 
  stat_cor(label.x.npc = 0.6, label.y.npc = 0.05) + 
  labs(x = expression(bold("PCR concentration (ng/"*mu*"L)")), y = "Clean sequence count") + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     #legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold'),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/clean_reads_count_vs_PCRconc_by_heat.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

g <- ggplot(dat, aes(PCRConc, Seq_count_clean, color = Extraction_method)) + 
  geom_point() + 
  scale_x_log10() + 
  scale_y_log10() +
  geom_smooth(method = "lm") + 
  stat_regline_equation(label.x.npc = 0.6,label.y.npc = 0.15) + 
  stat_cor(label.x.npc = 0.6, label.y.npc = 0.05) + 
  labs(x = expression(bold("PCR concentration (ng/"*mu*"L)")), y = "Clean sequence count") + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     #legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold'),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/clean_reads_count_vs_PCRconc_by_extract_method.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

```

### DNAConc vs PCRConc
```{r}
dat <- phylo_decontam_clean@sam_data@.Data %>% as.data.frame()
names(dat) <- phylo_decontam_clean@sam_data@names
dat$SampleID <- phylo_decontam_clean@sam_data@row.names
dat$Seq_count_clean <- sample_sums(phylo_decontam_clean)
dat <- dat %>% filter(!is.na(Country), Seq_count_clean >= depth)

g <- ggplot(dat, aes(DNA_Conc_ng_uL, PCRConc, color = heat_inactivated)) + 
  geom_point() + 
  scale_x_log10() + 
  scale_y_log10() +
  geom_smooth(method = "lm") + 
  stat_regline_equation(label.x.npc = 0.6,label.y.npc = 0.15) + 
  stat_cor(label.x.npc = 0.6, label.y.npc = 0.05) + 
  labs(x = expression(bold("DNA concentration (ng/"*mu*"L)")), y = expression(bold("PCR concentration (ng/"*mu*"L)"))) + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     #legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold'),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/DNAConc_vs_PCRconc_by_heat.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

g <- ggplot(dat, aes(DNA_Conc_ng_uL, PCRConc, color = Extraction_method)) + 
  geom_point() + 
  scale_x_log10() + 
  scale_y_log10() +
  geom_smooth(method = "lm") + 
  stat_regline_equation(label.x.npc = 0.6,label.y.npc = 0.15) + 
  stat_cor(label.x.npc = 0.6, label.y.npc = 0.05) + 
  labs(x = expression(bold("DNA concentration (ng/"*mu*"L)")), y = expression(bold("PCR concentration (ng/"*mu*"L)"))) + 
  theme_bw() + theme(#aspect.ratio = 0.8,
                     #legend.position="none",
                     panel.background = element_rect(fill = NA),
                     strip.text = element_text(size=12,color="black",face='bold'),
                     axis.title = element_text(size=12,color="black",face='bold'),
                     axis.text = element_text(size=12,color="black",face='bold'),
                     axis.text.x = element_text(size=12,color="black",face='bold'),
                     axis.text.y = element_text(size=12,color="black",face='bold'),
                     legend.text = element_text(size=12,color="black",face='bold'),
                     legend.title = element_text(size=12,color="black",face='bold'),
                     title = element_text(size=12,color="black",face='bold'))
ggsave(filename = "~/Dropbox/Rutgers/20220606_RUN41_milk/Data/output/DNAConc_vs_PCRconc_by_extract_method.pdf", 
       device = cairo_pdf, plot = g, width = 7, height = 5, units = "in", dpi = 300)

```




